src/IBM.Cloud.SDK.Core/Service/IBMService.cs

```diff
@@ -71,247 +66,54 @@ protected string Endpoint
-                iamClientId: !string.IsNullOrEmpty(ClientId) ? ClientId : null,
-                iamClientSecret: !string.IsNullOrEmpty(ClientSecret) ? ClientSecret : null
-                );
-
-            BasicAuthConfig basicAuthConfig = new BasicAuthConfig(
-                username: !string.IsNullOrEmpty(UserName) ? UserName : null,
-                password: !string.IsNullOrEmpty(Password) ? Password : null
-                );
-
-            if (!string.IsNullOrEmpty(ServiceUrl))
-                Endpoint = ServiceUrl;
-
-            if (!string.IsNullOrEmpty(Icp4dAccessToken) && !string.IsNullOrEmpty(Icp4dUrl))
-            {
-                SetAuthenticator(icp4dConfig);
-            }
-
-            if (!string.IsNullOrEmpty(ApiKey))
-            {
-                SetAuthenticator(iamConfig);
-            }
-
-            if (!string.IsNullOrEmpty(UserName) && !string.IsNullOrEmpty(Password))
-            {
-                SetAuthenticator(basicAuthConfig);
-            }
-        }
-
-        protected IBMService(string serviceName, string url)
+        protected IBMService(string serviceName, string defaultEndpoint)
         {
             ServiceName = serviceName;
-            Client = new IBMHttpClient(url, UserName, Password);
-
-            if (!string.IsNullOrEmpty(Endpoint))
-                Endpoint = url;
-            //TODO: verificar como iremos obter de um arquivo json por injeção de dependencia
-            //ApiKey = CredentialUtils.GetApiKey(serviceName);
-            //Endpoint = CredentialUtils.GetApiUrl(serviceName);
+            Client = new IBMHttpClient();
+            authenticator = ConfigBasedAuthenticatorFactory.GetAuthenticator(serviceName);
+            var creds = CredentialUtils.GetServiceProperties(serviceName);
+            creds.TryGetValue("URL", out string url);
+            SetEndpoint(string.IsNullOrEmpty(url) ? defaultEndpoint : url);
         }
 
         protected IBMService(string serviceName, string url, IClient httpClient)
         {
-            ServiceName = serviceName;
-            Client = httpClient;
+            SkipAuthentication = true;
+            ServiceName = serviceName;
+            Client = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
 
             if (!string.IsNullOrEmpty(Endpoint))
                 Endpoint = url;
-
-            //TODO: verificar como iremos obter de um arquivo json por injeção de dependencia
-            //ApiKey = CredentialUtils.GetApiKey(serviceName);
-            //Endpoint = CredentialUtils.GetApiUrl(serviceName);
         }
 
-        protected IBMService(string serviceName, IAuthenticatorConfig authenticatorConfig)
+        protected IBMService(string serviceName, Authenticator authenticator)
         {
-            if (authenticatorConfig.AuthenticationType == Authenticator.AuthtypeIcp4d)
+            if (authenticator.AuthenticationType == Authenticator.AuthtypeCp4d)

```

Comment:
Is this just a convenience to always disable SSL when users are authenticating with CP4D? I'm kind of leaning towards letting them just do that if they need.. Also for consistency, the merged Java PR doesn't have this behavior.

Reply:
Good catch, this was so `DisableSslVerification` from authentication trickles down to the service. I kept meaning to remove this!